python/lsst/ts/observatory/control/mock/atcs_mock.py

```diff
@@ -0,0 +1,572 @@
+# This file is part of ts_observatory_control.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+
+__all__ = ["ATCSMock"]
+
+import asyncio
+
+import numpy as np
+
+import astropy.units as u
+from astropy.time import Time
+from astropy.coordinates import EarthLocation, Angle
+
+from lsst.ts import salobj
+from lsst.ts.idl.enums import ATDome, ATPneumatics, ATMCS
+
+LONG_TIMEOUT = 30
+HEARTBEAT_INTERVAL = 1  # seconds
+
+
+class ATCSMock:
+    """ Mock the behavior of the combined components that make out ATTCS.
+
+    This is useful for unit testing.
```

Comment:
Would it make any sense to use the real CSCs (run in simulation mode) when such a CSC exists?

The advantages I see are:
* it is a lot less code to maintain. Your code will need to be updated every time the XML changes, or the expected behavior of the CSC. I have had a bad time with this already.
* It is much more realistic.

I suppose a big disadvantage is more dependencies.

Reply:
I'll give it some more though... maybe in the future... since only a subset of them have usable simulators, I'm a bit torn about it.

Reply:
That's fair. I suspect once you have rewritten your simple simulators a few times to track changes in the CSC you may change your mind, but at some point things should stabilize enough that it's not such a problem. Maybe we are there.